src/tests/unit_tests.c

```diff
@@ -0,0 +1,2116 @@
+				if (cp != buf)
+					inc_failed_test();
+			} else if (pfn) {
+				char *cp = (char*) pfn(buf, "012345ABCd99900AbC0", i, pad);
+				if (cp != buf)
+					inc_failed_test();
+			} else {
+				int n = nfn(buf, "012345ABCd99900AbC0", i);
+				if (n != strlen(buf))
+					inc_failed_test();
+			}
+			if (!case_type) {
+				if (strncmp(buf, "012345ABCd99900AbC0", check_len)) {
+					if (pfn && !strncmp(buf, pad_chk, check_len))
+						; // OK, pad check. The buffer 'filled', and emitted padding characters.
+					else
+						inc_failed_test();
+				}
+			} else {
+				if (strncasecmp(buf, "012345ABCd99900AbC0", check_len))
+					inc_failed_test();
+				if (case_type == 1 && i > 7 && !strncmp(buf, "012345ABCd99900AbC0", check_len))
+					inc_failed_test();
+				if (case_type == 2 && i > 10 && !strncmp(buf, "012345ABCd99900AbC0", check_len))
+					inc_failed_test();
+			}
+			if (null_check &&  buf[null_check])
+				inc_failed_test();
+			if (failed)
+				printf("Failed! Function: %s(buf,\"012345ABCd99900AbC0\", %d); data: %s\n", Results.test_name, i, hex(buf, i));
+			free(buf);
+		}
+	}
+}
+/*
+ * this function writes 'special' files. The files have each line be fully
+ * self describing. lines ALL start with  "line %04d  ", where the number is
+ * the line number. Then there will be some line which have the string [NULL]\0
+ * added after that signature. These test a known NULL bug (where the following
+ * line was skipped.  Lines that do not have the 'NULL\n' will instead have a
+ * space.  Then there is a ':' followed by a rotating set of characters. The
+ * length of each line is random, from 24 to 18999 bytes long. This will make
+ * sure that there are tests where we get a 'full' line (i.e. the size of our
+ * buffer is smaller than the length of data, BUT it also assures that we
+ * get data which is LONGER than our buffer.  This will test both fgetl and
+ * fgetll, which handle things differently. in fgetl, we truncate the line
+ * returned, to fit the buffer, throw away the remainder, and the next read
+ * MUST be at the start of the next line.  For fgetll, we instead allocate
+ * a different buffer, larger than the input buffer, read the whole line.
+ * the only case fgetll will return a truncated line, is if there is a NULL
+ * byte in the line.  We test 6000 lines, this allows the testing function
+ * to test 3 different buffer length (normal, small, huge), which will also
+ * help make sure there are enough buffer large enough, and buffer too small
+ * lines to test things well.  Each file will be 6000 lines long.
+ *
+ *  NOTE, one thing not tested, is files > 2gb and files > 4gb. This testing
+ *  was left out on purpose.  These unit tests are written to be run on the
+ *  CI's, and not knowing the limits on the CI's, I did not want to have
+ *  13+GB of test file data written. The tests will test everything other
+ *  than 64 bit file seeking, and 64 bit file telling.  However, seek/tell
+ *  is not part of fgetl or fgetll.
+ */
```

Comment:
This comment is properly formatted! 🎉

Reply:
Most should now 'fit'  There are still a few over 80 columns (with 8 byte tab stops), but very few.  There are some expressions and function calls longer, but I have cleaned things up quite a bit.